File: src/domain/repositories/FolderDateRepository.ts
import type { FolderDate } from '@/domain/entities/FolderDate'

export interface CreateFolderDateInput {
  folderId: string
  dateKey: string
  note: string
  createdById: string
  createdByName: string
}

export interface FolderDateRepository {
  getById(id: string): Promise<FolderDate | null>
  listByFolder(folderId: string): Promise<FolderDate[]>
  findByFolderAndDateKey(
    folderId: string,
    dateKey: string,
  ): Promise<FolderDate | null>
  create(input: CreateFolderDateInput): Promise<FolderDate>
  incrementImageCount(dateId: string, delta: number): Promise<void>
  delete(id: string): Promise<void>
  deleteAllByFolder(folderId: string): Promise<void>
}

File: consorcio/lib/domain/repositories/folder_date_repository.dart
import '../entities/folder_date.dart';

class CreateFolderDateInput {
  const CreateFolderDateInput({
    required this.folderId,
    required this.dateKey,
    required this.note,
    required this.createdById,
    required this.createdByName,
  });

  final String folderId;
  final String dateKey;
  final String note;
  final String createdById;
  final String createdByName;
}

abstract class FolderDateRepository {
  Future<FolderDate?> getById(String id);
  Future<List<FolderDate>> listByFolder(String folderId);
  Future<FolderDate?> findByFolderAndDateKey(String folderId, String dateKey);
  Future<FolderDate> create(CreateFolderDateInput input);
  Future<void> incrementImageCount(String dateId, int delta);
}
